Gems/Terrain/Code/Tests/TerrainSurfaceGradientListTests.cpp

```diff
@@ -37,81 +31,80 @@ namespace UnitTest
+        }
 
-            m_entity->Init();
+        UnitTest::MockTerrainLayerSpawnerComponent* AddRequiredComponentsToEntity(AZ::Entity* entity)
+        {
+            auto layerSpawnerComponent = entity->CreateComponent<UnitTest::MockTerrainLayerSpawnerComponent>();
+            m_app.RegisterComponentDescriptor(layerSpawnerComponent->CreateDescriptor());
 
-            m_gradientEntity1 = AZStd::make_unique<AZ::Entity>();
-            ASSERT_TRUE(m_gradientEntity1);
+            return layerSpawnerComponent;
+        }
+    };
 
-            m_gradientEntity1->Init();
+    TEST_F(TerrainSurfaceGradientListTest, SurfaceGradientMissingRequirementsActivateFails)
+    {
+        auto entity = CreateEntity();
 
-            m_gradientEntity2 = AZStd::make_unique<AZ::Entity>();
-            ASSERT_TRUE(m_gradientEntity2);
+        auto terrainSurfaceGradientListComponent = entity->CreateComponent<Terrain::TerrainSurfaceGradientListComponent>();
+        m_app.RegisterComponentDescriptor(terrainSurfaceGradientListComponent->CreateDescriptor());
 
-            m_gradientEntity2->Init();
-        }
+        const AZ::Entity::DependencySortOutcome sortOutcome = entity->EvaluateDependenciesGetDetails();
+        EXPECT_FALSE(sortOutcome.IsSuccess());
 
-        void AddSurfaceGradientListToEntities()
-        {
-            m_layerSpawnerComponent = m_entity->CreateComponent<UnitTest::MockTerrainLayerSpawnerComponent>();
-            m_app.RegisterComponentDescriptor(m_layerSpawnerComponent->CreateDescriptor());
+        entity.reset();
+    }
 
-            Terrain::TerrainSurfaceGradientListConfig config;
+    TEST_F(TerrainSurfaceGradientListTest, SurfaceGradientActivateSuccess)
+    {
+        auto entity = CreateEntity();
 
-            Terrain::TerrainSurfaceGradientMapping mapping1;
-            mapping1.m_gradientEntityId = m_gradientEntity1->GetId();
-            mapping1.m_surfaceTag = SurfaceData::SurfaceTag(surfaceTag1);
-            config.m_gradientSurfaceMappings.emplace_back(mapping1);
+        AddRequiredComponentsToEntity(entity.get());
 
-            Terrain::TerrainSurfaceGradientMapping mapping2;
-            mapping2.m_gradientEntityId = m_gradientEntity2->GetId();
-            mapping2.m_surfaceTag = SurfaceData::SurfaceTag(surfaceTag2);
-            config.m_gradientSurfaceMappings.emplace_back(mapping2);
+        auto terrainSurfaceGradientListComponent = entity->CreateComponent<Terrain::TerrainSurfaceGradientListComponent>();
+        m_app.RegisterComponentDescriptor(terrainSurfaceGradientListComponent->CreateDescriptor());
+
+        entity->Activate();
 
-            Terrain::TerrainSurfaceGradientListComponent* terrainSurfaceGradientListComponent =
-                m_entity->CreateComponent<Terrain::TerrainSurfaceGradientListComponent>(config);
-            m_app.RegisterComponentDescriptor(terrainSurfaceGradientListComponent->CreateDescriptor());
-        }
-    };
+
+
+        entity.reset();
+    }
```

Comment:
This test is not doing any checks, missing to see if entity is active:

````
EXPECT_EQ(entity->GetState(), AZ::Entity::State::Active);
````

Reply:
> 
> 
> This test is not doing any checks, missing to see if entity is active:
> 
> ```
> EXPECT_EQ(entity->GetState(), AZ::Entity::State::Active);
> ```

Oops, thanks.